### DISCUSSION TITLE
How to set exact width of columns in react-table version 8 which will not change in any condition?

### DISCUSSION THREAD

**Discussion by yashwant-raut** [TanStack/table#4707] [Q&A]
I'm using a table with an expand option while working on a react table. With react table version 7, which I used to make this table, I was able to adjust the width of each table column using the width property and the useBlockLayout hook. Now I'm switching to react-table version 8, or tan table, for performance reasons.

In version 8, I can create a table with the expand option and also set the width using the size option, but whatever size I enter is not applied exactly to columns. For example, if I enter 118 as size, it takes 120 or 110 in pixels. Additionally, if I click the expand all button and the size of the first column changes, it reduces the size of other columns.

Any width I specify in the older version appears to be applied to the table, and columns' sizes are preserved if the size of an outer contained window changes.

1. My implementation of react table version 7 with expand option and fixed columns width [React table in version 7 with fixed column width](https://codesandbox.io/s/relaxed-nightingale-f1f0jz?file=/src/App.js)
2. My implementation of react table version 8 with expand option and variable columns width [Link to my implementation](https://codesandbox.io/p/sandbox/vigorous-sanderson-zlklzz?file=%2Fsrc%2Fmain.tsx&selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A210%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A210%7D%5D)

In conclusion, I want to give the table fixed column sizes that shouldn't alter under any circumstances.

In First implementation I have set width and that exact width is applied in version 7.

But in version 8 of table given size is not applied to column table. And my constraint is I don't want to increase the width and height of table Container Class which is holding this table

Please refer both the implementation of mine and please tell me what is I am missing in tan table implementation because of that it is not maintaining the width of column
